tests/integration/integration_python_test.go

```diff
@@ -400,3 +403,30 @@ func TestGetResourcePython(t *testing.T) {
 		AllowEmptyPreviewChanges: true,
 	})
 }
+
+// Regresses https://github.com/pulumi/pulumi/issues/6471
+func TestAutomaticVenvCreation(t *testing.T) {
+	// Do not use itnegration.ProgramTest to avoid automatic venv
+	// handling by test harness; we actually are testing venv
+	// handilng by the pulumi CLI itself.
```

Comment:
Nit

```suggestion
	// Do not use integration.ProgramTest to avoid automatic venv
	// handling by test harness; we actually are testing venv
	// handling by the pulumi CLI itself.
```